### PR TITLE
Add generic representer

### PR DESCRIPTION
**Pull request by iHiD**
I **think** this will work pretty easily.

### CODE REVIEW

File: app/commands/submission/representation/process.rb [hunk 1]
```diff
@@ -4,99 +4,25 @@ class Submission::Representation::Process
   initialize_with :tooling_job
 
   def call
```
Comment (iHiD) [inline app/commands/submission/representation/process.rb:6]: All the code below this is moved into `process_results` below

File: app/commands/submission/representation/init.rb [hunk 2]
```diff
@@ -11,19 +11,27 @@ def initialize(submission, type: :submission, git_sha: nil, run_in_background: f
...
+    queue_job!.tap do
+      update_status!
+    end
+  end
+
+  private
+  attr_reader :submission, :type, :git_sha, :run_in_background
+
+  def queue_job!
+    return Submission::Representation::GenerateBasic.defer(submission) unless track.has_representer?
```
Comment (iHiD) [inline app/commands/submission/representation/init.rb:23]: This is the change to this file.

File: app/commands/submission/representation/process_results.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
+class Submission::Representation::ProcessResults
```
Comment (iHiD) [inline app/commands/submission/representation/process_results.rb:1]: This is the code that was deleted above.

File: app/commands/submission/representation/process.rb (hunk 1 above)
Comment (iHiD) [reply app/commands/submission/representation/process.rb:6]: This file is now effectively responsible for taking a tooling job and extracting the relevant data to pass to `processresults`

File: test/commands/submission/representation/generate_basic_test.rb [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+  test "normalises ast correctly" do
+    filename_1 = "start.rb"
+    code_1 = "foo eeee"
+    filename_2 = "end.rb"
+    code_2 = "bar wwww \n aaaa"
+
+    # Check things are normalised correctly
+    digest = Submission::Representation.digest_ast("#{filename_1}\n#{code_1.gsub(/\s/,
+      '')}\n\n#{filename_2}\n#{code_2.gsub(/\s/, '')}")
```
Comment (ErikSchierboom) [inline test/commands/submission/representation/generate_basic_test.rb]: I wonder if using a plain multiline string without the interpolated values and gsubbing, which makes reading it a bit harder I feel:

```suggestion
# Check things are normalised correctly
    expected_ast = <<~AST
      start.rb
      fooeeee

      end.rb
      barwwwwaaaa
    AST
    digest = Submission::Representation.digest_ast(expected_ast.chomp)
```

Comment (iHiD) [reply test/commands/submission/representation/generate_basic_test.rb]: Yep.

**Pull request merged**